### PR TITLE
Modify k8s Ingress Controller tutorial to support cluster

### PR DESCRIPTION
**Pull request by samutamm**
Update `doc/ingresscontroller.md` tutorial to include cluster support
- Change *Deployment* to *StatefulSet*
- Use *PersistentVolumes* for disk access

Refactor Helm charts. `helm-charts` contains three chart directories
- `helm-charts/easegress`: general Helm chart for installing single instance or multi-node Easegress cluster
- `helm-charts/ingress-controller`: install Easegress as an ingress controller
- `helm-charts/easegress-lib-chart`: [Helm library](https://helm.sh/docs/topics/library_charts/) for some of the common parts of the other charts. It would be possible to move even more, currently duplicated, yaml definition to this common library. However it might require more yaml templating logic, which might make the charts less readable.

### CODE REVIEW

File: doc/ingresscontroller.md [hunk 1]
```diff
@@ -247,7 +434,306 @@ Version: 2.0.0
...
+kind: StorageClass
+metadata:
+  name: easegress-storage
+provisioner: kubernetes.io/no-provisioner
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer
+```
+Now here's how we add all three cluster member URLs to *ConfigMap*:
+
+```yaml
```
Comment (localvar) [inline doc/ingresscontroller.md:546]:
```suggestion
```

File: doc/ingresscontroller.md [hunk 2]
```diff
@@ -247,7 +434,306 @@ Version: 2.0.0
...
+    api-addr: 0.0.0.0:2381
+    data-dir: /opt/eg-data/data
+    wal-dir: ""
+    cpu-profile-file: ""
+    memory-profile-file: ""
+    log-dir: /opt/eg-data/log
+    debug: false
+  controller.yaml: |
+    ...
+---
```
Comment (localvar) [inline doc/ingresscontroller.md:574]:
```suggestion
```

File: doc/ingresscontroller.md [hunk 3]
```diff
@@ -247,7 +434,306 @@ Version: 2.0.0
...
```
Comment (localvar) [inline doc/ingresscontroller.md:638]:
```suggestion
```

File: doc/ingresscontroller.md [hunk 4]
```diff
@@ -247,7 +434,306 @@ Version: 2.0.0
...
+        name: easegress-cluster-cm
+---
+```
+
+Applying the deployment to the cluster creates two pods with prefix `easegress-secondary-`. We can again log into one of pods and see new *secondary* members.
+
+```bash
+# query admin port with egctl
+egctl --server {NODE_IP}:31255 member list |grep " name"
+# => prints
```
Comment (localvar) [inline doc/ingresscontroller.md:732]: propose to move the comments out of the shell script, some user may think they are commands to execute.

File: helm-charts/easegress/templates/Deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
+
```
Comment (localvar) [inline helm-charts/easegress/templates/Deployment.yaml:1]:
```suggestion
```

File: helm-charts/easegress-lib-chart/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: library
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
```
Comment (localvar) [inline helm-charts/easegress-lib-chart/Chart.yaml:18]: propose to align this version number with Easegress version

**Review by localvar — APPROVED** [review]

**Review by suchen-sci — COMMENTED** [review]
Hao has just reorganized the doc. Please pull the newest doc and update it to your pr. For example, "k8s_ingress_controller.md" should be "k8s-ingress-controller.md" and "doc/ingresscontroller.md" should in "doc/reference"....

**Comment by samutamm** [comment]
> Hao has just reorganized the doc. Please pull the newest doc and update it to your pr. For example, "k8s_ingress_controller.md" should be "k8s-ingress-controller.md" and "doc/ingresscontroller.md" should in "doc/reference"....

I rebased the main branch, it's now up to date @suchen-sci

**Review by suchen-sci — APPROVED** [review]
LGTM!

**Pull request merged**